### ISSUE TITLE
CRASH -[FCKeyChainStore setData:forKey:genericAttribute:label:comment:error:]

### ISSUE THREAD

**Issue by failynsedik** [freshdesk/freshchat-ios#99]
Hi,

I am experiencing crashes and it's happening for quite a number of times. Please see screenshot and log. Thank you.
<img width="493" alt="multiple_crash" src="https://user-images.githubusercontent.com/52100901/78219990-44e5b080-74f3-11ea-910f-2394918b5e01.png">

**Log from Crashlytics:**
```
com.apple.root.user-initiated-qos
0  ???                            0x1c680900 (Missing)
1  ???                            0x1c6806e1 (Missing)
2  ???                            0x1c5a9af1 (Missing)
3  ???                            0x1c5a9ebf (Missing)
4  ???                            0x1c774821 (Missing)
5  Security                       0x1d267721 securityd_message_with_reply_sync + 154
6  Security                       0x1d285aaf SecItemRawUpdate + 246
7  Security                       0x1d2859b5 __SecItemUpdateWithError_block_invoke + 88
8  Security                       0x1d286b09 __SecItemAuthDoQuery_block_invoke + 204
9  Security                       0x1d28462f SecItemAuthDo + 902
10 Security                       0x1d285083 SecItemAuthDoQuery + 420
11 Security                       0x1d28591f SecItemUpdateWithError + 442
12 Security                       0x1d283d63 SecOSStatusWith + 42
13 Security                       0x1d285755 SecItemUpdate + 64
14 LalaFood-User                  0x3d2bd7 -[FCKeyChainStore setData:forKey:genericAttribute:label:comment:error:] + 3714007
15 LalaFood-User                  0x3d2623 -[FCKeyChainStore setData:forKey:] + 3712547
16 LalaFood-User                  0x3dc987 -[FCSecureStore setObject:forKey:] + 3754375
17 LalaFood-User                  0x42a8b1 __42-[Freshchat updateConfig:andRegisterUser:]_block_invoke + 4073649
18 ???                            0x1c595797 (Missing)
19 ???                            0x1c5a42db (Missing)
20 ???                            0x1c5a400f (Missing)
21 ???                            0x1c74a87d (Missing)
22 ???                            0x1c74a45c (Missing)
```

Edit: This is mostly happening on **iPhone 5 devices** that are on **iOS 10.x.x**. Also, I released the app using **Xcode 11.4**.

**Comment by harish-kumarc** [comment]
Thanks for the details @failynsedik , Which version of iOS SDK creating this crash?

**Comment by bestwnh** [comment]
@failynsedik it maybe a Xcode11.4 issue. You can also check this issue https://github.com/firebase/firebase-ios-sdk/issues/5233. We better use Xcode 11.3.1 for now. Because event if the Freshchat can fix it, some other frameworks or our code may also trigger the crash.

**Comment by failynsedik** [comment]
Hello @bestwnh. Thanks for that and sorry for the late response. Have you tried to upload using Xcode 11.5? Do you know if Apple (Xcode team) has fixed it?

**Comment by bestwnh** [comment]
@failynsedik Yes. The issue fix by building the project with Xcode 11.5.
